packages/checkout/components/store-notices-container/store-notices.tsx

```diff
@@ -87,70 +90,71 @@ const StoreNotices = ( {
 			className={ classnames( className, 'wc-block-components-notices' ) }
 		>
 			{ nonDismissibleNotices.map( ( notice ) => (
-				<Notice
+				<StoreNotice
 					key={ notice.id + '-' + notice.context }
-					className={ classnames(
-						'wc-block-components-notices__notice',
-						getClassNameFromStatus( notice.status )
-					) }
 					{ ...notice }
 				>
 					{ sanitizeHTML( decodeEntities( notice.content ) ) }
-				</Notice>
+				</StoreNotice>
 			) ) }
 			{ Object.entries( dismissibleNoticeGroups ).map(
 				( [ status, noticeGroup ] ) => {
 					if ( ! noticeGroup.length ) {
 						return null;
 					}
-					const uniqueNotices = noticeGroup.filter(
-						(
-							notice: Notice,
-							noticeIndex: number,
-							noticesArray: Notice[]
-						) =>
-							noticesArray.findIndex(
-								( _notice: Notice ) =>
-									_notice.content === notice.content
-							) === noticeIndex
-					);
-					return (
-						<Notice
-							key={ `store-notice-${ status }` }
-							className={ classnames(
-								'wc-block-components-notices__notice',
-								getClassNameFromStatus( status )
-							) }
-							onRemove={ () => {
-								noticeGroup.forEach( ( notice ) => {
-									removeNotice( notice.id, notice.context );
-								} );
-							} }
+					const uniqueNotices = noticeGroup
+						.filter(
+							(
+								notice: NoticeType,
+								noticeIndex: number,
+								noticesArray: NoticeType[]
+							) =>
+								noticesArray.findIndex(
+									( _notice: NoticeType ) =>
+										_notice.content === notice.content
+								) === noticeIndex
+						)
+						.map( ( notice ) => ( {
+							...notice,
+							content: sanitizeHTML(
+								decodeEntities( notice.content )
+							),
+						} ) );
+					const noticeProps = {
```

Comment:
There is an error when rendering `StoreNotice` because TS incorrectly assumes that `status` is a string, rather than one of the allowed strings.

To get around it, you could type `noticeProps` like:

```
Omit< NoticeBannerProps, 'children' > & {
  key: string;
}
```